changelog/fragments/rescue-reconciliation.yaml

```diff
@@ -0,0 +1,8 @@
+entries:
+  - description: >
+      Added checks for rescued tasks during reconciliation
+      to avoid adding rescued tasks to the list of failures.
+
+    kind: "addition"
+
+    breaking: false
```

Comment:
The same suggestions of: https://github.com/operator-framework/operator-sdk/pull/3727/files#